mantidproject/mantid
# Improvements to SliceMD algorithm

`SliceMD` now uses the specified binning parameters to set the `SplitTopInto` parameter, rather than `SplitInto`.  `SplitInto` is set to the same value as the input workspace. This results in a far more sensible box structure for densely populated, multidimensional slices, and greatly reduces their size in memory.
The algorithm also now reports (in the log) the correct number of events in the output workspace.

The section on "Splitting Parameters" was updated in the user documentation for `SliceMD`.
The release notes were also updated (VATES CLI section of framework.rst).

**To test:**
- Grab the archive of test data from `smb://olympic/babylon5/Scratch/Matthew%20Jones/Fe_data.zip` and unzip it.
- Replace the path in the following python script with the location you extracted the data at.
- Run the script, it uses SliceMD via the CutMD workflow algorithm. Note that it will take a few minutes to read in the data files. Prior to the changes in this PR this use of CutMD would quickly use all available memory and crash MantidPlot. Now it should complete successfully and the resultant workspace (out_md) should be ~700MB, the script will print the size.

``` python
import os
import numpy as np

#Make MD mega file (like gen_sqw / accumulate_sqw)
psi_arr=np.arange(0.0,92.0,2)  # 46.0
run_arr=range(15052,15098)  # range(15052,15075)

input_runs = list()
psi = list()
gs = list()
gl = list()
#note that gl, gs etc are not mandtory inputs. If we do not specify them then they are set to zero.

for i in range(len(psi_arr)):
    source_file = '/REPLACE/WITH/PATH/TO/Fe_data/map'+str(run_arr[i])+'_ei400.nxspe'
    input_runs.append(source_file)
    psi.append(float(psi_arr[i]))
    gl.append(0.0)
    gs.append(0.0)

print(psi)
print(input_runs)

md_ws = CreateMD(input_runs, Emode='Direct', Alatt=[2.87,2.87,2.87], Angdeg=[ 90, 90, 90], u=[1, 0, 0,], v=[0,1,0], Psi=psi, Gl=gl, Gs=gs, EFix=400.0)

from mantid.api import Projection
# Create an identity projection
proj_id = Projection([1,1,0], [-1,1,0], [0,0,1])
proj_id.setType(0, 'r')
proj_id.setType(1, 'r')
proj_id.setType(2, 'r')
ws_id = proj_id.createWorkspace()

out_md = CutMD(md_ws, Projection=ws_id, PBins=([-3,0.05,3],[-3,0.05,3],[-0.1,0.1],[0,4,360]), NoPix=False)
print ("Size of output workspace is "+str(out_md.getMemorySize()/(1024*1024))+"MB")
```

Fixes #15652.



---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
